lisa/features/infiniband.py

```diff
@@ -171,7 +171,34 @@ def get_pkey(self) -> str:
         cat = self._node.tools[Cat]
         return cat.read(f"/sys/class/infiniband/{ib_device_name}/ports/1/pkeys/0")
 
-    def setup_rdma(self) -> None:  # noqa: C901
+    def setup_rdma(self, install_ofed: bool = True) -> None:
+        if install_ofed:
+            self.install_ofed()
+
+        node = self._node
+        # Turn off firewall
+        firewall = node.tools[Firewall]
+        firewall.stop()
+        # Disable SELinux
+        sed = node.tools[Sed]
+        sed.substitute(
+            regexp="SELINUX=enforcing",
+            replacement="SELINUX=disabled",
+            file="/etc/selinux/config",
+            sudo=True,
+        )
+
+        if isinstance(node.os, Ubuntu) and node.os.information.version > "18.4.0":
+            node.tools[Sed].substitute(
```

Comment:
Add comments above to explain the purpose.

Reply:
updated